src/main/java/core/basesyntax/service/impl/ParserImpl.java

```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.model.Operation;
+import core.basesyntax.service.Parser;
+
+public class ParserImpl implements Parser {
+    private static final String DELIMITER = ",";
+    private static final int OPERATION_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int QUANTITY_INDEX = 2;
+
+    @Override
+    public FruitTransaction parse(String data) {
+        if (!data.matches("^[brsp],.*")) {
+            return null;
+        }
+        String[] strings = data.split(DELIMITER);
+        String fruitName = strings[NAME_INDEX];
+        int quantity = Integer.parseInt(strings[QUANTITY_INDEX]);
+        Operation operation = Operation
+                .fromCode(strings[OPERATION_INDEX]);
```

Comment:
Let's wrap this in a try-catch block since we might encounter trouble parsing malformed input. We should throw a runtime exception with a message indicating that this transaction is in the wrong format, along with the transaction details

Reply:
Also, move the logic of iterating the list of transactions from Main to this class, let's keep Main short